src/blog/first-100k-lines.vmd

```diff
@@ -0,0 +1,415 @@
+
+
+Let's say you have a `id_to_account HashMap<ID, UserAccount>`. Unfortunately, to find a user by name, you have to loop through the entire map, because it's keyed by ID, not by name.
+
+So then you add a separate `name_to_account HashMap<str, UserAccount>`, and you try to keep these two maps in sync. However, if you accidentally remove an account from only one, you now have a *data inconsistency.*
+
+
+After you add your normal assertions, also consider periodically calling a `sanityCheck` function:
+
+```vale
+func sanityCheck(
+    id_to_account &HashMap<ID, UserAccount>,
+    name_to_account &HashMap<str, UserAccount>) {
+  assert(id_to_account.size() == name_to_account_map.size());
+  assert(name_to_account.map(x => x.id) == id_to_account.keys());
+  // ... and even more assertions!
+}
+```
+
+
+
+In one case, we have an [80-line sanity check function](https://github.com/ValeLang/Vale/blob/f4ff0d4385484bced63a181c80bb2f81a48ab83f/Frontend/Solver/src/dev/vale/solver/OptimizedSolverState.scala) to check that all the state in the generics solver is consistent.
+
+
+When I worked on [Earth](https://earth.google.com/web/), I made a 200-line sanityCheck function run before and after every click, which made sure the application was in a sane state. It saved _countless_ hours of debugging.
+
+
+Lean hard on this technique, it will serve you well.
+
+
+
+# That's all!
+
+Thanks for visiting, and I wish you the best of luck in your first 100,000 lines!
+
+In the coming weeks, I'll be writing more about our "region borrow checker" which helps eliminate Vale's memory safety overhead, so subscribe to our [RSS feed](https://verdagon.dev/rss.xml) or the [r/Vale](https://reddit.com/r/vale) subreddit, and come hang out in the [Vale discord](https://discord.gg/SNB8yGH).
+
+
+If you found this interesting or entertaining, please consider sponsoring me:
+
+<center>
+  <a href="https://github.com/sponsors/ValeLang" class="donate-button">
+     <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-heart icon-sponsor mr-1 color-fg-sponsors">
+        <path fill-rule="evenodd" d="M4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.565 20.565 0 008 13.393a20.561 20.561 0 003.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.75.75 0 01-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5zM8 14.25l-.345.666-.002-.001-.006-.003-.018-.01a7.643 7.643 0 01-.31-.17 22.075 22.075 0 01-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.08 22.08 0 01-3.744 2.584l-.018.01-.006.003h-.002L8 14.25zm0 0l.345.666a.752.752 0 01-.69 0L8 14.25z"></path>
+     </svg>
+     Sponsor me on GitHub!
+  </a>
+</center>
+
+With your help, I can write this kind of nonsense more often!
+
+
+
+<slice new-color="afterword"/>
```

Comment:
feel free to ignore everything after here, its just standard copy-paste for all articles now